### DISCUSSION TITLE
How to add more languages to the Code Editor?

### DISCUSSION THREAD

**Discussion by OwainJ** [leekelleher/umbraco-contentment#390] [Question] [answered]
On the code editor, and in the docs, it states:

> Please note, by default, Umbraco ships a streamlined set of programming language modes and themes.
> 
> If you would like to add more modes and themes, you can do this by [downloading the latest pre-packaged version of the Ace editor](https://github.com/ajaxorg/ace-builds/releases) and copy any of the mode-* or theme-* files from the src-min-noconflict folder over to the ~/umbraco/lib/ace-builds/src-min-noconflict/ folder in you Umbraco installation.
> 
> Once you've done this, you can reload the Data Type screen, and the new programming language modes and themes will appear in the dropdown options for the fields above.

But that doesn't seem possible anymore because since (I think v10?) Umbraco has removed all of the backoffice code from the `umbraco` folder that lives at the project root. 

Therefore, how can more languages be added in Umbraco v13?

**Comment by leekelleher** [comment, marked as answer]
Hi @OwainJ, it should still work, but I suspect that the folder needs to be in the web-root,
e.g. `wwwroot\umbraco\lib\ace-builds\src-min-noconflict`.
